File: go/chunk_array.go
func chunk(arr []int, step int) []int {
  a, o := make([]int, len(arr)), []int{}
  copy(a, arr)
  if step < 0 {
    for i, j := 0, len(a)-1; i < j; i, j = i+1, j-1 {
        a[i], a[j] = a[j], a[i]
    }
    step*=-1
  }
  for i := 0 ; i < len(a) ; i+=step {
    o = append(o, a[i])
  }
  return o
}

File: rust/chunk_array.rs
fn chunk(arr: Vec<i32>, h: i32) -> Vec<i32> {
    let l=arr.len();
    let mut a:Vec<i32> = Vec::new();
    let mut o:Vec<i32> = Vec::new();
    let mut x=h;
    if x < 0 {
        for i in (0..l).rev() {
            a.push(arr[i]);
        }
        x*=-1;
    } else {
        for i in 0..l {
            a.push(arr[i]);
        }
    }
    let step=x as usize;
    for i in (0..a.len()).step_by(step) {
        o.push(a[i]);
    }
    o
}
